torchgeo/datasets/quakeset.py

```diff
@@ -0,0 +1,289 @@
+# Copyright (c) Microsoft Corporation. All rights reserved.
+# Licensed under the MIT License.
+
+"""QuakeSet dataset."""
+
+import os
+from collections.abc import Callable
+from typing import Any, cast
+
+import matplotlib.pyplot as plt
+import numpy as np
+import torch
+from matplotlib.figure import Figure
+from torch import Tensor
+
+from .geo import NonGeoDataset
+from .utils import DatasetNotFoundError, download_url, percentile_normalization
+
+
+class QuakeSet(NonGeoDataset):
+    """QuakeSet dataset.
+
+    `QuakeSet <https://huggingface.co/datasets/DarthReca/quakeset>`__
+    is a dataset for Earthquake Change Detection and Magnitude Estimation and is used
+    for the Seismic Monitoring and Analysis (SMAC) ECML-PKDD 2024 Discovery Challenge.
+
+    Dataset features:
+
+    * Sentinel-1 SAR imagery
+    * before/pre/post imagery of areas affected by earthquakes
```

Comment:
What is the difference between before and pre?

Reply:
The before+pre = unaffected area, pre+post = earthquake affected area. You can think of before+pre as a hard negative because it's of the same location of the earthquake but a "change" pair prior to the event.

Reply:
Let's add this to the docstring in some form.

Reply:
done